test/util/range.cpp

```diff
@@ -213,6 +213,21 @@ void test_pointers()
     BOOST_CHECK(bgr::at(r2, 5) == 5);
 }
 
+void test_const_pointers()
```

Comment:
I was not aware that the `unit-test` rule helps in that. This has to be investigated indeed.

Regarding the advantages: I think you benefited from it in another PR: you downloaded boost, ran `b2 headers` and were ready to go and test....